src/metrics/instance_definitions_test.go

```diff
@@ -7,30 +7,68 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func Test_generateInstanceDefinitions90(t *testing.T) {
-	v := semver.MustParse("9.0.0")
-	queryDefinitions := generateInstanceDefinitions(&v)
+func Test_generateInstanceDefinitions(t *testing.T) {
+	tests := []struct {
+		name            string
+		version         string
+		expectedQueries []*QueryDefinition
+		errorExpected   bool
+	}{
+		{
+			name:            "PostgreSQL 9.0",
+			version:         "9.0.0",
+			expectedQueries: []*QueryDefinition{instanceDefinitionBase},
+			errorExpected:   false,
+		},
+		{
+			name:            "PostgreSQL 9.1",
+			version:         "9.1.0",
+			expectedQueries: []*QueryDefinition{instanceDefinitionBase, instanceDefinition91},
+			errorExpected:   false,
+		},
+		{
+			name:            "PostgreSQL 9.2",
+			version:         "9.2.0",
+			expectedQueries: []*QueryDefinition{instanceDefinitionBase, instanceDefinition91, instanceDefinition92},
+			errorExpected:   false,
+		},
+		{
+			name:            "PostgreSQL 10.2",
+			version:         "10.2.0",
+			expectedQueries: []*QueryDefinition{instanceDefinitionBase, instanceDefinition91, instanceDefinition92},
+			errorExpected:   false,
+		},
+		{
+			name:            "PostgreSQL 16.4",
+			version:         "16.4.2",
+			expectedQueries: []*QueryDefinition{instanceDefinitionBase, instanceDefinition91, instanceDefinition92},
+			errorExpected:   false,
+		},
+		{
+			name:            "PostgreSQL 17.0",
+			version:         "17.0.0",
+			expectedQueries: []*QueryDefinition{instanceDefinitionBase170, instanceDefinition170, instanceDefinitionInputOutput170},
+			errorExpected:   false,
+		},
+		{
+			name:            "PostgreSQL 17.5 out of order",
+			version:         "17.5.0",
+			expectedQueries: []*QueryDefinition{instanceDefinition170, instanceDefinitionInputOutput170, instanceDefinitionBase170},
+			errorExpected:   true,
+		},
```

Comment:
Nit: it is not actually an error, is it? (the function being tested is working fine, what is not right is the `expectedQueries` value in the test itself). I think we can remove this test case (or update it to have te proper expectations) and simplify the test by removing the `errorExpected` flag.

Reply:
Handled this as you described. Moved out of order to a new test to simplify things.